src/libtiled/properties.h

```diff
@@ -43,7 +43,7 @@ struct FilePath {
 };
 
 struct ObjectRef {
-    int id = 0;
+    int id;
```

Comment:
Not sure about this being only a C++11 problem. The code is compiled as C++14 everywhere and compiled fine with both GCC and clang (are you on Visual Studio?). That said, we can't just remove this, since we need to initialize somehow. Let's just have a constructor then.

Reply:
https://en.cppreference.com/w/cpp/language/aggregate_initialization

> no default member initializers (since C++11, until C++14)

So I guess that includes C++14?

Also, it's supposed to be zero-initialized by the aggregate constructor:

> If the number of initializer clauses is less than the number of members or initializer list is completely empty, the remaining members are value-initialized. If a member of a reference type is one of these remaining members, the program is ill-formed.

And on https://en.cppreference.com/w/cpp/language/value_initialization

> 1) if T is a class type with at least one user-provided constructor...
1) if T is a class type with no default constructor...
2) if T is a class type with a default constructor that is neither user-provided nor deleted (that is, it may be a class with an implicitly-defined or defaulted default constructor), the object is zero-initialized and then it is default-initialized if it has a non-trivial default constructor;
3) if T is an array type...
4) otherwise, the object is zero-initialized.

So we don't need to provide a constructor to ensure id is set to 0.

Reply:
Also, no, it didn't compile fine on MINGW. One of the automated tests failed with that error.

Reply:
> Also, no, it didn't compile fine on MINGW. One of the automated tests failed with that error.

Whoops, I guess I checked the CI builds after they had already run with your fix...

Thanks for pointing out the value-initialization for initializer lists! That does mean that without this explicit initialization, we need to avoid writing `ObjectRef()` (currently see three occurrences).

Reply:
> That does mean that without this explicit initialization, we need to avoid writing `ObjectRef()` (currently see three occurrences).

So, actually according to https://en.cppreference.com/w/cpp/language/value_initialization, the form `T()` also does value-initialization, so there is no need to change those occurrences.